CASMINST-4903 - csi should better handle invalid input in the paddle file

spillerc-hpe opened the pull request:
### Summary and Scope

- Fixes: [CASMINST-4903](https://jira-pro.its.hpecorp.net:8443/browse/CASMINST-4903)

#### Issue Type

- Bugfix Pull Request

Better handle the case where the CANU generated ccj file may contain errors caused by bad SHCD formatting. For example the rack elevation "U" location being uppercase when the code expects lowercase.

### Prerequisites

- [X] I have included documentation in my PR (or it is not required)
- [X] I tested this on internal system (if yes, please include results or a description of the test)

Tested on surtur, command now functions as expected
```
surtur-ncn-m001-pit:/var/www/ephemeral/prep/cspiller # csi config shcd surtur-full-paddle.json -ANS
2022/07/20 09:17:26 Created "switch_metadata.csv" from SHCD data
2022/07/20 09:17:26 Created "application_node_config.yaml" from SHCD data
2022/07/20 09:17:26 Created "ncn_metadata.csv" from SHCD data
```

Previously it failed

```
surtur-ncn-m001-pit:/var/www/ephemeral/prep/cspiller # csi config shcd surtur-full-paddle.json -ANS
2022/07/20 09:22:28 strconv.Atoi: parsing "U14": invalid syntax
```
 
### Idempotency
 
### Risks and Mitigations

thiagonache commented:
Please, don't forget to write a test for this case! @jacobsalmela @spillerc-hpe @gbaker-hpe

rustydb reviewed (changes requested):
Please add a test case.

spillerc-hpe commented:
> Please add a test case.

I have fixed the broken `TestCreateSwitchMetadata` test, tests tend to pass when what you read in as your expected data is what you just generated.

I've also added a new test which fails when run with the old version of this code and now passes.

With code fix
```
=== RUN   TestCreateSwitchMetadataWithMalformedCCJ
2022/08/04 13:50:11 Created "switch_metadata.csv" from SHCD data
--- PASS: TestCreateSwitchMetadataWithMalformedCCJ (0.00s)
```

Without code fix
```
=== RUN   TestCreateSwitchMetadataWithMalformedCCJ
2022/08/04 13:49:10 strconv.Atoi: parsing "U14": invalid syntax
FAIL	github.com/Cray-HPE/cray-site-init/cmd	0.703s
```

Please note `TestCreateSwitchMetadata` and `TestCreateSwitchMetadataWithMalformedCCJ` cannot run in parallel because they interfere with each other's output files.

thiagonache commented:
great work, folks!
